src/main/java/com/percero/agents/auth/services/AuthService2.java

```diff
@@ -0,0 +1,691 @@
+                    if (entityImpl != null) {
+                        RelationshipImplementation relImpl = entityImpl.findRelationshipImplementationBySourceVarName(IUserIdentifier.USER_ANCHOR_FIELD_NAME);
+                        if (relImpl != null) {
+                            relImpl.sourceMappedField.getSetter().invoke(nextUserIdentifier, result);
+                            if (isNewObject) {
+                                syncAgentService.systemCreateObject((IPerceroObject) nextUserIdentifier, null);
+                            }
+                            else {
+                                syncAgentService.systemPutObject((IPerceroObject) nextUserIdentifier, null, null, null, true);
+                            }
+                        }
+                        else {
+                            logger.warn("No UserAnchor found for IUserIdentifier class " + nextUserIdentifier.getClass().getCanonicalName());
+                        }
+                    }
+                }
+
+                if (result != null) {
+                    setupUserRoles(user, serviceUser);
+                    return result;
+                }
+            }
+
+        } catch (IllegalArgumentException e) {
+            logger.error(e.getMessage(), e);
+        } catch (IllegalAccessException e) {
+            logger.error(e.getMessage(), e);
+        } catch (InvocationTargetException e) {
+            logger.error(e.getMessage(), e);
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+        } finally {
+            if (s != null && s.isOpen())
+                s.close();
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    public void setupUserRoles(User user, ServiceUser serviceUser) throws Exception {
+        Session s = appSessionFactory.openSession();
+
+        try {
+            EntityImplementation userAnchorEI = getUserAnchorEntityImplementation();
+            EntityImplementation userRoleEI = getUserRoleEntityImplementation();
+
+            RelationshipImplementation userAnchorRelImpl = userRoleEI.findRelationshipImplementationBySourceVarName(IUserRole.USER_ANCHOR_FIELD_NAME);
+
+            String userAnchorQueryString = "SELECT ua FROM " + userAnchorEI.mappedClass.tableName + " ua WHERE ua.userId=:userId";
+            Query userAnchorQuery = s.createQuery(userAnchorQueryString);
+            userAnchorQuery.setString("userId", user.getID());
+
+            IUserAnchor userAnchor = (IUserAnchor) userAnchorQuery.uniqueResult();
+
+            //IUserRoleA userRoleAnnotation = getUserRoleAnnotation(userRoleClass);
+            String personRoleQueryString = "SELECT ur FROM " + userRoleEI.mappedClass.tableName
+                    + " ur WHERE ur." + userAnchorRelImpl.sourceMappedField.getField().getName() + " IN (SELECT ua FROM " + userAnchorEI.mappedClass.tableName + " ua WHERE ua.userId=:userId)";
```

Comment:
Same comment about the `userId` field name being configurable via the UserAnchor annotation.